cli/actions/run_test_action.go

```diff
@@ -154,3 +168,51 @@ func (a runTestAction) runTest(ctx context.Context, testID string) (openapi.Test
 
 	return *run, nil
 }
+
+func (a runTestAction) waitForResult(ctx context.Context, testId string, testRunId string) (openapi.TestRun, error) {
+	var testRun openapi.TestRun
+	var lastError error
+	var wg sync.WaitGroup
+	wg.Add(1)
+	ticker := time.NewTicker(1 * time.Second) // TODO: make this configurable
+	go func() {
+		for {
+			select {
+			case <-ticker.C:
+				isReady, readyTestRun, err := a.isTestReady(ctx, testId, testRunId)
+				if err != nil {
+					lastError = err
+					wg.Done()
+					return
+				}
+
+				if isReady {
+					testRun = readyTestRun
+					wg.Done()
+					return
+				}
+			}
+		}
+	}()
+	wg.Wait()
+
+	if lastError != nil {
+		return openapi.TestRun{}, lastError
+	}
+
+	return testRun, nil
+}
+
+func (a runTestAction) isTestReady(ctx context.Context, testId string, testRunId string) (bool, openapi.TestRun, error) {
```

Comment:
nit: in this case, it might make more sense to return an `*openapi.TestRun` and return `nil` instead of zeroed `openapi.TestRun`